services/stats/stats.go

```diff
@@ -160,6 +160,9 @@ func Setup() {
 				taggedClientsMapLock.Unlock()
 			}
 		}
+		taggedClientsMapLock.Lock()
```

Comment:
Shouldn't these statements be outside the new goroutine and right after statsd.New?